test/tests/functional/pbs_reservations.py

```diff
@@ -883,3 +883,128 @@ def test_fail_confirm_resv_message(self):
+        self.server.expect(RESV, a, id=rid)
+
+        a = {ATTR_q: rid_q, ATTR_J: '1-4'}
+        j2 = Job(TEST_USER, attrs=a)
+        j2.set_sleep_time(10)
+        jid2 = self.server.submit(j2)
+        subjid = []
+        for i in range(1, 5):
+            subjid.append(j.create_subjob_id(jid2, i))
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=20)
+        self.server.expect(JOB, {'job_state': 'B'}, jid2)
+        self.server.expect(JOB, {'job_state=R': 1}, count=True,
+                           id=jid2, extend='t')
+        self.server.expect(JOB, {'job_state=Q': 3}, count=True,
+                           extend='t', id=jid2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[0])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=subjid[1])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=subjid[2])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=subjid[3])
+        self.server.deljob(subjid[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[1])
+        # Wait for reservation to delete from server
+        msg = "Que;" + rid_q + ";deleted at request of pbs_server@"
+        self.server.log_match(msg, starttime=now, interval=10)
+        # Check status of the sub-job using qstat -fx once job completes
+        self.server.expect(JOB, {'job_state': 'F', 'Exit_status': '271'},
+                           extend='x', attrop=PTL_AND, id=subjid[0])
+        self.server.expect(JOB, {'job_state': 'F', 'Exit_status': '0'},
+                           extend='x', attrop=PTL_AND, id=subjid[3])
+
+        # Submit a advance reservation (R2) and an array job to the reservation
+        # once reservation confirmed
+
+        r = Reservation(TEST_USER)
+        now = int(time.time())
+        a = {'Resource_List.select': '1:ncpus=4',
+             'reserve_start': now + 20,
+             'reserve_end': now + 180}
+        r.set_attributes(a)
+        rid = self.server.submit(r)
+        rid_q = rid.split('.')[0]
+        a = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, a, id=rid)
+
+        a = {ATTR_q: rid_q, ATTR_J: '1-4'}
+        j2 = Job(TEST_USER, attrs=a)
+        j2.set_sleep_time(60)
+        jid2 = self.server.submit(j2)
+        subjid = []
+        for i in range(1, 5):
+            subjid.append(j.create_subjob_id(jid2, i))
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=20)
+        self.server.expect(JOB, {'job_state': 'B'}, jid2)
+        self.server.expect(JOB, {'job_state=R': 4}, count=True,
+                           id=jid2, extend='t')
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[1])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[2])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid[3])
```

Comment:
Since you only have 5 subjobs, instead of checking 1-4 being in state R, you could just check for subjob 5 being in state Q.  You just checked that there are 4 running jobs.